mapie/calibration.py

```diff
@@ -0,0 +1,548 @@
+import warnings
+from typing import Dict, Optional, Tuple, Union, cast
+
+import numpy as np
+from sklearn.base import BaseEstimator, ClassifierMixin, RegressorMixin, clone
+from sklearn.calibration import _SigmoidCalibration
+from sklearn.isotonic import IsotonicRegression
+from sklearn.utils import check_random_state
+from sklearn.utils.multiclass import type_of_target
+from sklearn.utils.validation import (_check_y, _num_samples, check_is_fitted,
+                                      indexable)
+
+from ._typing import ArrayLike, NDArray
+from .utils import (check_binary_zero_one, check_cv,
+                    check_estimator_classification,
+                    check_estimator_fit_predict, check_n_features_in,
+                    check_null_weight, fit_estimator, get_calib_set)
+
+
+class MapieCalibrator(BaseEstimator, ClassifierMixin):
+    """
+    Calibration for multi-class problems.
+
+    This class performs a post-processing step on the output of a classifier
+    more representative of the confidence of the prediction, closer to the
+    metric of accuracy.
+
+    Parameters
+    ----------
+    estimator : Optional[ClassifierMixin]
+        Any classifier with scikit-learn API
+        (i.e. with fit, predict, and predict_proba methods), by default None.
+        If ``None``, estimator defaults to a ``LogisticRegression`` instance.
+
+    method: Optional[str]
+        Method to choose for prediction interval estimates.
+        Choose among:
+
+        - "top_label", performs a calibration procedure on the class with
+           highest probability score.
+
+        By default "top_label".
+
+    estimator : Optional[Union[str, RegressorMixin]]
+        Any estimator with scikit-learn API
+        (i.e. with fit, predict, and predict_proba methods), by default None.
+        If ``None``, estimator defaults to a string ``sigmoid`` instance.
+
+        By default ``None``.
+
+    cv: Optional[str]
```

Comment:
do you offer several methods for sklearn compliancy? I think it would be more natural to only offer prefit

Reply:
I am not sure about sklearn compliancy in this case, but the split method is very much required, otherwise this means we would only allow using estimators that have been prefit.